src/pages/Overview/FiltersAndSorts.ts

```diff
@@ -1,16 +1,56 @@
-import { FILTER_ACTION_APPEND, FilterType, FilterValue } from '../../types/Filters';
-import { FAILURE, DEGRADED, HEALTHY } from '../../types/Health';
+import { ActiveFilter, FILTER_ACTION_APPEND, FilterTypeWithFilter, FilterValue } from '../../types/Filters';
+import { DEGRADED, FAILURE, HEALTHY } from '../../types/Health';
 import { NamespaceInfo } from './NamespaceInfo';
 import { SortField } from '../../types/SortFilters';
+import { MTLSStatuses } from '../../types/TLSStatus';
 
 export namespace FiltersAndSorts {
-  export const nameFilter: FilterType = {
+  export const nameFilter: FilterTypeWithFilter<NamespaceInfo> = {
     id: 'namespace_search',
     title: 'Name',
     placeholder: 'Filter by Name',
     filterType: 'text',
     action: FILTER_ACTION_APPEND,
-    filterValues: []
+    filterValues: [],
+    filter: (namespaces: NamespaceInfo[], filters: ActiveFilter[]) => {
+      return namespaces.filter(ns => filters.some(f => ns.name.includes(f.value)));
+    }
+  };
+
+  export const mtlsValues: FilterValue[] = [
+    {
+      id: 'enabled',
+      title: 'Enabled'
+    },
+    {
+      id: 'partiallyEnabled',
+      title: 'Partially Enabled'
+    },
+    {
+      id: 'notEnabled',
+      title: 'Not Enabled'
+    }
+  ];
+
+  const statusMap = new Map<string, string>([
+    [MTLSStatuses.ENABLED, 'Enabled'],
+    [MTLSStatuses.PARTIALLY, 'Partially Enabled'],
+    [MTLSStatuses.NOT_ENABLED, 'Not Enabled'],
+    [MTLSStatuses.DISABLED, 'Not Enabled']
+  ]);
+
+  export const mtlsFilter: FilterTypeWithFilter<NamespaceInfo> = {
+    id: 'mtls',
+    title: 'mTLS status',
+    placeholder: 'Filter by mTLS status',
+    filterType: 'select',
+    action: FILTER_ACTION_APPEND,
+    filterValues: mtlsValues,
+    filter: (namespaces: NamespaceInfo[], filters: ActiveFilter[]) => {
+      return namespaces.filter(
+        ns => ns.tlsStatus && filters.some(f => statusMap.get(ns.tlsStatus!.status) === f.value)
```

Comment:
Because of `ns.tlsStatus!.status`, could it be a problem if the filtering function is called while not all asynchronous data have been fetched? (in which case `ns.tlsStatus` would be undefined i guess)

Reply:
Good point. I think that I won't be a problem. Having `ns.tlsStatus` at the beginning of the statement will prevent the execution of the `some` expression, so I guess that non undefined access error will be rose.